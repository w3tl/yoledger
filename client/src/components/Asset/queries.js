import gql from 'graphql-tag';
import fragments from './fragments';

export const LIST_QUERY = gql`
query AssetsQuery {
  assets: accounts(type: ASSET) {
    ...AssetFormAccount
  }
}
${fragments.asset}
`;

export const ADD_MUTATION = gql`
mutation addAccount($input: AddAccountInput!) {
  addAccount(input: $input) {
    account {
      ...AssetFormAccount
    }
  }
}
${fragments.asset}
`;
