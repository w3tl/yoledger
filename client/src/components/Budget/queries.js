import gql from 'graphql-tag';

export const BODY_QUERY = gql`
query BudgetPeriodsQuery($dateStart: Date!, $count: Int) {
  budgets(dateStart: $dateStart, count: $count) {
    periods
    accounts {
      name
    }
  }
}
`;

export const COLUMN_QUERY = gql`
query BudgetQuery($date: Date!) {
  budget(date: $date) {
    id
    account {
      name
    }
    amount
  }
}
`;

export const UPDATE_MUTATION = gql`
mutation upsertBudget($input: UpsertBudgetInput!) {
  upsertBudget(input: $input) {
    success
    allocation {
      id
      account { id name }
      amount
    }
  }
}
`;
