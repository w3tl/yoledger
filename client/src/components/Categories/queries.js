import gql from 'graphql-tag';
import fragments from './fragments';

export const LIST_QUERY = gql`
query CategoriesQuery {
  expenses: accounts(type: EXPENSE) {
    ...Category
  }
  incomes: accounts(type: INCOME) {
    ...Category
  }
}
${fragments.category}
`;

export const ADD_MUTATION = gql`
mutation addAccount($input: AddAccountInput!) {
  addAccount(input: $input) {
    account {
      ...Category
    }
  }
}
${fragments.category}
`;
