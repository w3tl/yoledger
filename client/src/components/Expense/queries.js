import gql from 'graphql-tag';
import fragments from './fragments';

export const LIST_QUERY = gql`
query ExpensesQuery {
  accounts(type: EXPENSE) {
    ...ExpenseAccount
  }
}
${fragments.expense}
`;

export const FORM_QUERY = gql`
query ExpenseQuery($id: ID!) {
  account(id: $id) {
    ...ExpenseAccount
  }
}
${fragments.expense}
`;

export const ADD_MUTATION = gql`
mutation addAccount($input: AddAccountInput!) {
  addAccount(input: $input) {
    account {
      ...ExpenseAccount
    }
  }
}
${fragments.expense}
`;
