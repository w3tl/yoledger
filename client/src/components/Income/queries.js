import gql from 'graphql-tag';
import fragments from './fragments';

export const LIST_QUERY = gql`
query IncomesQuery {
  accounts(type: INCOME) {
    ...IncomeAccount
  }
}
${fragments.income}
`;

export const FORM_QUERY = gql`
query IncomeQuery($id: ID!) {
  account(id: $id) {
    ...IncomeAccount
  }
}
${fragments.income}
`;

export const ADD_MUTATION = gql`
mutation addAccount($input: AddAccountInput!) {
  addAccount(input: $input) {
    account {
      ...IncomeAccount
    }
  }
}
${fragments.income}
`;
