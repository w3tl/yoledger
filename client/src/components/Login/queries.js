import gql from 'graphql-tag';

export const SIGNIN_MUTATION = gql`
mutation signin($login: String!, $password: String!) {
  signin(login: $login, password: $password) {
    token
  }
}
`;

export const SIGNOUT_MUTATION = gql`
mutation signout($token: String) {
  signout(token: $token) {
    success
  }
}
`;
