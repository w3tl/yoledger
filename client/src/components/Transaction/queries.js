import gql from 'graphql-tag';
import fragments from './fragments';

export const FORM_QUERY = gql`
query TransactionQuery($id: ID!) {
  transaction(id: $id) {
    ...TransactionFormTransaction
  }
}
${fragments.transaction}
`;

export const LIST_QUERY = gql`
query TransactionsQuery($dateStart: Date!, $dateEnd: Date, $page: Int, $itemsPerPage: Int) {
  transactions(dateStart: $dateStart, dateEnd: $dateEnd, page: $page, itemsPerPage: $itemsPerPage) {
    ...TransactionFormTransaction
  }
}
${fragments.transaction}
`;

export const ADD_MUTATION = gql`
mutation addTransaction($input: AddTransactionInput!) {
  addTransaction(input: $input) {
    transaction {
      ...TransactionFormTransaction
    }
  }
}
${fragments.transaction}
`;

export const DELETE_MUTATION = gql`
mutation deleteTransaction($id: ID!) {
  deleteTransaction(id: $id) {
    success
    id
  }
}
`;

export const UPDATE_MUTATION = gql`
mutation updateTransaction($id: ID!, $input: UpdateTransactionInput!) {
  updateTransaction(id: $id, input: $input) {
    transaction {
      ...TransactionFormTransaction
    }
  }
}
${fragments.transaction}
`;

export const GET_PAGE_SETTINGS = gql`
query {
  settings @client {
    transaction {
      period
      date
    }
  }
}
`;

export const CHANGE_PERIOD = gql`
  mutation changePeriod($period: Period, $date: String) {
    changeTransactionPagePeriod(period: $period, date: $date) @client {
      period
      date
    }
  }
`;
