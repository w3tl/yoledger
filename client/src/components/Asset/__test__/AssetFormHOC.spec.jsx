import React from 'react';
import { mount } from 'enzyme';
import { LIST_QUERY } from '../queries';
import RawAssetForm, { updateAfterAdd } from '../AssetFormHOC';
import {
  wait, withProvider, testLoadingState, client,
} from '../../testHelpers';


const AssetForm = withProvider((props) => {
  props.client.writeQuery({ query: LIST_QUERY, data: { assets: [] } });
  return <RawAssetForm />;
});

describe('AssetFormHOC', () => {
  testLoadingState(<AssetForm />);

  describe('updateAfterAdd', () => {
    it('should update assets cache', () => {
      const writeQuery = jest.fn();
      const cache = {
        readQuery: jest.fn().mockReturnValue({ assets: [] }),
        writeQuery,
      };
      const data = { addAccount: { account: { type: 'ASSET' } } };
      updateAfterAdd(cache, { data });
      expect(writeQuery.mock.calls[0][0].data).toMatchSnapshot('writeQuery');
    });
  });

  it('should contain props', async () => {
    const wrapper = mount(<AssetForm />);
    await wait();
    wrapper.update();
    const form = wrapper.find('AssetForm');
    expect(form.prop('asset').id).toBeDefined();
  });

  it('should successfully add asset', async () => {
    const wrapper = mount(<AssetForm />);
    await wait();
    wrapper.update();
    wrapper.find('input[name="name"]').first().simulate('change', { target: { value: 'new account' } });
    wrapper.find('form').simulate('submit');
    await wait(1);
    wrapper.update();
    const { assets } = client.readQuery({ query: LIST_QUERY });
    expect(assets).toHaveLength(1);
  });
});
