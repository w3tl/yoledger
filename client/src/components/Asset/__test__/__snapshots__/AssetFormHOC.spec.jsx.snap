// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetFormHOC updateAfterAdd should update assets cache: writeQuery 1`] = `
Object {
  "assets": Array [
    Object {
      "type": "ASSET",
    },
  ],
}
`;
